packages/payments/src/channel-manager.ts

```diff
@@ -388,6 +399,47 @@ export class ChannelManager implements ChannelManagementAPI {
     };
   }
 
+  /**
+   *
+   * @param objectiveIds objectiveIds to sync on
+   * @param message initial message to send to indexer
+   *
+   * 1. exchanges messages until outbox is empty
+   * 2. if any objectives have not finished, waits for a period of time,
+   *    and then sync the in progress objectives
+   */
+  private async ensureObjectives(
+    objectives: DBObjective[],
+    initialMessage: Outgoing
+  ): Promise<ChannelResult[]> {
+    const remaining = new Map(objectives.map((o) => [o.objectiveId, o]));
+
+    objectives.map(({objectiveId}) =>
+      this.wallet.on('objectiveSucceeded', (o) => {
+        if (o.objectiveId === objectiveId) {
+          remaining.delete(objectiveId);
+        }
+      })
+    );
+
+    const results = await this.exchangeMessagesUntilOutboxIsEmpty(initialMessage);
+    const latestResult = new Map(results.map((c) => [c.channelId, c]));
+
+    for (const retryTimeoutMs of this.backoffIntervals) {
+      if (remaining.size === 0) return Array.from(latestResult.values());
```

Comment:
I am probably missing something, but how is `remaining` updated in this loop?

Reply:
They are deleted in L420 above
```
        if (o.objectiveId === objectiveId) {
          remaining.delete(objectiveId);
        }
```